Attempting to compile source gives unstable features errors

```zsh
error[E0658]: use of unstable library feature 'stdsimd'
  --> /home/$USER/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.3/src/backend/vector/ifma/field.rs:26:5
   |
26 |     _mm256_madd52lo_epu64(z.into(), x.into(), y.into()).into()
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
   = help: add `#![feature(stdsimd)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'stdsimd'
  --> /home/$USER/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.3/src/backend/vector/ifma/field.rs:25:9
   |
25 |     use core::arch::x86_64::_mm256_madd52lo_epu64;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
   = help: add `#![feature(stdsimd)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'stdsimd'
  --> /home/$USER/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.3/src/backend/vector/ifma/field.rs:34:5
   |
34 |     _mm256_madd52hi_epu64(z.into(), x.into(), y.into()).into()
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
   = help: add `#![feature(stdsimd)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'stdsimd'
  --> /home/$USER/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.3/src/backend/vector/ifma/field.rs:33:9
   |
33 |     use core::arch::x86_64::_mm256_madd52hi_epu64;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #48556 <https://github.com/rust-lang/rust/issues/48556> for more information
   = help: add `#![feature(stdsimd)]` to the crate attributes to enable

error[E0635]: unknown feature `stdarch_x86_avx512`
  --> /home/$USER/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.3/src/lib.rs:19:13
   |
19 |     feature(stdarch_x86_avx512)
   |             ^^^^^^^^^^^^^^^^^^

   Compiling growable-bloom-filter v2.1.0
   Compiling flate2 v1.0.30
   Compiling rusqlite v0.30.0
   Compiling event-listener v5.3.1
   Compiling makepad-live-compiler v0.5.0 (https://github.com/kevinaboos/makepad?branch=rik_with_old_font_stack#149b4c28)
   Compiling tokio-rustls v0.26.0
   Compiling hyper-util v0.1.6
   Compiling matrix-sdk-store-encryption v0.7.0 (https://github.com/matrix-org/matrix-rust-sdk#11d5e568)
   Compiling deadpool-sync v0.1.4
Some errors have detailed explanations: E0635, E0658.
For more information about an error, try `rustc --explain E0635`.
error: could not compile `curve25519-dalek` (lib) due to 5 previous errors
warning: build failed, waiting for other jobs to finish...
```

Had tried initially by installing the nightly toolchain in my system despite the presence of [`rust_toolchain.toml`](https://github.com/project-robius/robrix/blob/main/rust-toolchain.toml) which would override what toolchain is used,
```zsh
$ rustup install nightly
$ rustup default nightly
```
removing and adding cargo as a component,
```zsh
$ rustup component remove cargo # Strangely this did not work as cargo was never added as a component initially
$ rustup component add cargo
```

Updating cargo and building from a clean directory
```zsh
$ cargo update
$ cargo clean
```

None of this worked, however. I then disabled the [rust-analyzer](https://github.com/rust-lang/rust-analyzer) extension in VScode and it seemed to work.
Interestingly, repeating the same steps on another machine, Linux again, did not allow compilation.
I'm not sure what the issue is here, perhaps something to do with the extension not using the toolchain that the .toml file lists here or some issue with rustup, but it really does make compiling this project more difficult than it should be...

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "nightly-2023-12-22"
+channel = "stable"

--- a/src/profile/user_profile.rs
+++ b/src/profile/user_profile.rs
@@ -1,7 +1,7 @@
 use std::{borrow::Cow, cell::RefCell, collections::{btree_map::Entry, BTreeMap}, ops::{Deref, DerefMut}, sync::Arc};
 use crossbeam_queue::SegQueue;
 use makepad_widgets::*;
-use matrix_sdk::{room::RoomMember, ruma::events::room::member::MembershipState, OwnedRoomId, OwnedUserId, UserId};
+use matrix_sdk::{room::RoomMember, ruma::{events::room::member::MembershipState, OwnedRoomId, OwnedUserId, UserId}};
 use crate::{
     shared::avatar::AvatarWidgetExt, sliding_sync::{get_client, submit_async_request, MatrixRequest}, utils
 };
@@ -57,6 +57,7 @@ impl UserProfileUpdate {
     /// only if the user_id and room_id match that of the update.
     ///
     /// Returns `true` if the update resulted in any actual content changes.
+    #[allow(unused)]
     fn apply_to_current_pane(&self, info: &mut UserProfilePaneInfo) -> bool {
         match self {
             UserProfileUpdate::Full { new_profile, room_id, room_member } => {

